src/index.js

```diff
@@ -1,31 +1,80 @@
 import './style.css';
 
 const toDoContainer = document.getElementById('todolist');
+const itemInput = document.getElementById('input-item');
 
-const toDoTasks = [{
-  description: 'Sleep',
-  completed: false,
-  index: 1,
-},
-{
-  description: 'Play',
-  completed: false,
-  index: 2,
-},
-{
-  description: 'Eat',
-  completed: false,
-  index: 3,
-}];
-
-toDoTasks.sort((a, b) => a.index - b.index);
-
-toDoTasks.forEach((element) => {
-  toDoContainer.innerHTML += `
-        <ul class="list-items">
-            <li><input type="checkbox" name="item" class="check-for-items"></li>
-            <li>${element.description}</li>
-            <li class="ellipse"><i class="fa-solid fa-ellipsis-vertical"></i></li>
-        </ul>
-        `;
+function Tasks(description, index) {
+  this.description = description;
+  this.completed = false;
+  this.index = index;
+}
+
+class Books {
+  constructor() {
+    this.toDoTasks = localStorage.books ? JSON.parse(localStorage.books) : [];
+  }
+
+    updateToDoList = () => {
+      toDoContainer.innerHTML = '';
+      for (let i = 0; i < this.toDoTasks.length; i += 1) {
+        toDoContainer.innerHTML += `
+    <div class="list-items">
+        <input type="checkbox" name="item" class="check-for-items" id="${this.toDoTasks[i].index}" onchange='ticked(${this.toDoTasks[i].index})'>
+        <input type="text" value="${this.toDoTasks[i].description}" class="todo-item-text">
+        <i class='material-icons delete-btn' onclick="removeItem(${this.toDoTasks[i].index})">delete</i>
+    </div>`;
+      }
+    }
+
+    addEntry = (book) => {
+      this.toDoTasks.push(book);
+      this.saveToDoList();
+      this.updateToDoList();
+    };
+
+    saveToDoList = () => {
+      localStorage.setItem('books', JSON.stringify(this.toDoTasks));
+    };
+
+    remove(ids) {
+      this.toDoTasks = this.toDoTasks.filter((element) => element.index !== ids);
+      this.renewIndex();
+      this.saveToDoList();
+      this.updateToDoList();
+    }
+
+    renewIndex = () => {
+      for (let i = 0; i < this.toDoTasks.length; i += 1) {
+        this.toDoTasks[i].index = i + 1;
+      }
+      this.saveToDoList();
+    }
+}
+
+const books = new Books();
+books.updateToDoList();
+
+itemInput.addEventListener('focusout', () => {
+  const newTask = new Tasks(itemInput.value, books.toDoTasks.length + 1);
+  books.addEntry(newTask);
+  itemInput.value = '';
+});
```

Comment:
- [ ] As suggested by the previous reviewer, Kindly consider implementing the edit function such that when a user edits a todo, the updated version is reflected both on the page and in the `localStorage`. This will ensure that the changes made by the user persist even when they refresh the page or revisit it at a later time. Therefore, kindly consider updating the `localStorage` along with updating the todo item on the page to maintain consistency between the page and the `localStorage`.